src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -1916,6 +1929,61 @@ withNextSrv c userId usedSrvs initUsed action = do
     writeTVar usedSrvs $! used'
   action srvAuth
 
+incSMPServerStat :: AgentClient -> UserId -> SMPServer -> (AgentSMPServerStats -> TVar Int) -> Int -> STM ()
+incSMPServerStat AgentClient {smpServersStats} userId srv sel n = do
```

Comment:
what's the case when n is not 1? we need to count blocks sent to the server, not individual commands.

Not sure what's the best place to do it - current approach is not correct. Probably it's best to either pass counter (TVar) to the client or to return from the client the number of sent blocks that is known after batching (and also some blocks can expire after sending).

Reply:
if I understand correctly, there's no batching possible where it's currently called. and protocol batching is only used for subscriptions and queue deletions.